### DISCUSSION TITLE
Regions somehow not working?

### DISCUSSION THREAD

**Discussion by moonwhaler** [Acly/krita-ai-diffusion#784] [Q&A] [answered]
Hello,

first off: this is a great tool and perfect for any inpainting needs, something which I only used the selection tools for (as I am faily new to Krita). Anyway, I just installed the new version, that supports regions and tried to follow the video (YouTube), but it seems to me that regions are not applied.

The "base prompt" is used, but I have the feeling that the regions are somewhat ignored as the things "drawn" into their specific "paint layers" are not applied (at all). Plus: it seems after "applying" the generated images not all groups got their respective "cut out" of the defined (generated) image.

I am using Krita 5.2.2 - no other plugins applied. And - all - components, workflows, controlnet models etc. installed. I installed Comfy using the included install variant (local managed server). The install of Comfy is about 3-4 days old.

![2024-06-11 23_43_27-](https://github.com/Acly/krita-ai-diffusion/assets/1736467/d72a5be2-c09e-4f62-864c-c14a9de21f3e)
![2024-06-11 23_44_23-Layers - Krita](https://github.com/Acly/krita-ai-diffusion/assets/1736467/d4644472-681e-4936-b2b9-de4ee351278d)
![2024-06-11 23_44_49-](https://github.com/Acly/krita-ai-diffusion/assets/1736467/00f15a5b-1643-4738-8136-23d36b941063)

Not sure what exactly I could have done differently. Thank you for any idea!

**Comment by Acly** [comment, marked as answer]
Hi,

I think it's likely the region for the ship is simply too small. Try making it bigger and see what happens.

If you look closely I actually encounter same problem in the video when trying to paint the chameleon. The prompt isn't used until I go to a higher resolution and zoom in on it.

Think of regions as somewhat coarse compositional tool, they must cover at least 2% of the image, and technically they are evaluated at small resolutions. It's also difficult for AI models to draw subjects that don't cover a signficant portion of the image, that's what they've been trained on. You can get around this limitation by generating the background first, then zooming in on your subject with a selection/inpaint.

If you're wondering about the bottom result layer (above white Background), that's because the white part of the image is not a region. If you make the Background a region you could describe the sky there, but it's optional and will just use the base prompt if you don't.

**Comment by moonwhaler** [comment]
Thank you for the detailed answer. I will try this as soon as I am home.

You mentioned 'zooming in'. Does your plugin transfer the zoomed in area, if I use the Krita zooming function - or is it another form of zoom?

Sorry for the question, I am still new to Krita.

Thank you!

**Reply by Acly** [reply]
No Krita's canvas zoom is independent and doesn't affect generation.

In Live mode generation is automatically restricted to an area around the active region. To get a similar effect with the regular generation UI you can draw a selection (just a rect is fine for this purpose), to generate only part of the image.

**Reply by moonwhaler** [reply]
Ah, so the box selection is like a virtual zoom then. I understand. I will try this in the evening and close this question. Thanks again!
